src/graphene.cpp

```diff
@@ -792,6 +792,91 @@ double CGrapheneBlockData::average(std::map<int64_t, uint64_t> &map)
     return (double)accum / map.size();
 }
 
+double CGrapheneBlockData::computTotalBandwidthSavingsInternal() EXCLUSIVE_LOCKS_REQUIRED(cs_graphenestats)
+{
+    AssertLockHeld(cs_graphenestats);
+
+    return double(nOriginalSize() - nGrapheneSize() - nTotalMemPoolInfoBytes());
+}
+
+double CGrapheneBlockData::compute24hInboundCompressionInternal() EXCLUSIVE_LOCKS_REQUIRED(cs_graphenestats)
+{
+    AssertLockHeld(cs_graphenestats);
+
+    expireStats(mapGrapheneBlocksInBound);
+
+    double nCompressionRate = 0;
+    uint64_t nGrapheneSizeTotal = 0;
+    uint64_t nOriginalSizeTotal = 0;
+    for (std::map<int64_t, std::pair<uint64_t, uint64_t> >::iterator mi = mapGrapheneBlocksInBound.begin();
```

Comment:
nit: IMO `for (auto|<type> m : ...) {...}` is more readable

Reply:
I hadn't changed the original algorithms beyond moving them into helper methods.   I can refactor this and any other locations where I touched the code.

Reply:
Fixed for all old style for loops in thinblock.cpp and graphene.cpp related to stats computations.

Reply:
Great!